cmd/skopeo/sync.go

```diff
@@ -604,19 +606,22 @@ func (opts *syncOptions) run(args []string, stdout io.Writer) error {
 			if err != nil {
 				return err
 			}
-
-			logrus.WithFields(logrus.Fields{
+			fromToFields := logrus.Fields{
 				"from": transports.ImageName(ref),
 				"to":   transports.ImageName(destRef),
-			}).Infof("Copying image ref %d/%d", counter+1, len(srcRepo.ImageRefs))
-
-			if err = retry.RetryIfNecessary(ctx, func() error {
-				_, err = copy.Image(ctx, policyContext, destRef, ref, &options)
-				return err
-			}, opts.retryOpts); err != nil {
-				return errors.Wrapf(err, "Error copying ref %q", transports.ImageName(ref))
 			}
-			imagesNumber++
+			logrus.WithFields(fromToFields).Infof("Copying image ref %d/%d", counter+1, len(srcRepo.ImageRefs))
+			if opts.dry {
+				logrus.WithFields(fromToFields).Info("Would have copied image, but `--dry` is specified")
+			} else {
+				if err = retry.RetryIfNecessary(ctx, func() error {
+					_, err = copy.Image(ctx, policyContext, destRef, ref, &options)
+					return err
+				}, opts.retryOpts); err != nil {
+					return errors.Wrapf(err, "Error copying ref %q", transports.ImageName(ref))
+				}
+				imagesNumber++
```

Comment:
Should this be not counted in the dry-run mode? AFAICS this will now say “Synced 0 images from $nonzero sources”, which is a bit strange.

Reply:
I'd argue it's nice to be explicit here of what _would_ have happened, but did not.

Reply:
So maybe change the summary message to say “Would have synced … images”? Or alternatively drop the `--dry-run` reference from all the other messages, and move it into this summary message?

(I know, tinkering with details, and I’m not even really around this week. Just things to possibly consider. If others think this is fine to merge as-is, that’s perfectly fine with me.)

Reply:
I think I just addressed the second approach in 7635815. Let me know what you think!

Reply:
I was thinking a warning at the end, because that’s right above the next CLI prompt; a warning at the beginning will scroll away in interactive use. It _is_ a good place for warnings in non interactive iterations, because it will be at the top of a log file, but I don’t think that’s our use: it makes little sense to me to have a `--dry-run` in something like `cron`, it seems likely to be used interactively when tuning the configuration.

Something like `In --dry-run mode, didn't sync %d images from %d sources` ?

OTOH I don’t feel strongly about this; what do others think?

Reply:
I could see both ways. I'm happy to add an additional mention of dry-run at the end, to be thorough.